smse/pipelines/image.py

```diff
@@ -22,43 +37,66 @@ class ImagePipeline(BasePipeline):
     def __init__(self, config: ImageConfig):
         super().__init__(config)
         self.config: ImageConfig = config
+        # Initialize the transform Pipeline.
+        self.transform_pipeline = self._create_transform_pipeline()
+
+    def _create_transform_pipeline(self) -> transforms.Compose:
+        """Create a transformation pipeline for image preprocessing."""
+        transform_list = [
+            transforms.Lambda(self._ensure_rgb),  # Ensure RGB Color mode.
+            transforms.ToTensor(),  # Convert PIL to Tensor
+        ]
+
+        if self.config.target_size:  # Resize images.
+            transform_list.append(transforms.Resize(self.config.target_size))
+
+        if self.config.normalize:  # Normalize images.
+            transform_list.append(transforms.Normalize(mean=self.config.mean, std=self.config.std))
+
+        return transforms.Compose(transform_list)
+
+    @staticmethod
+    def _ensure_rgb(image: Image.Image) -> Image.Image:
+        """Ensure image is in RGB mode."""
+        if image.mode != "RGB":
+            image = image.convert("RGB")
+        return image
 
     def load(self, input_path: Union[str, Path]) -> ImageT:
         """Load image from file"""
         try:
             import cv2  # type: ignore[import-not-found]
-
             image: ImageT = cv2.imread(str(input_path))
-
+            if image is None:
+                raise ValueError(f"Failed to load image: {input_path}")
             return image
         except ImportError:
             self.logger.warning("OpenCV not found, trying PIL")
             from PIL import Image  # type: ignore[import-not-found]
-
             return np.array(Image.open(str(input_path)))
 
-    def validate(self, data: Any) -> bool:
-        return isinstance(data, np.ndarray) and len(data.shape) in [2, 3]
+    def process(self, images: List[ImageT], device: Optional[torch.device] = None) -> torch.Tensor:
+        """
+        Preprocess a list of images using the transform pipeline.
 
-    def process(self, image: ImageT) -> ImageT:
-        """Preprocess image data"""
-        import cv2  # type: ignore[import-not-found]
+        Args:
+            images (list[PIL.Image | ImageT]): List of input images (either PIL or NumPy).
 
-        # Resize
-        image = cv2.resize(image, self.config.target_size)
+        Returns:
+            torch.Tensor: Preprocessed batch of images with shape [B, C, H, W].
+        """
+        if device is None:
+            device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
The base config already has a field called device which is set to `cpu` by default.